[Bug] CommandExecutor.get_process_status() returns OS-level process status instead of expected "running" status

## Problem/Motivation

The `test_get_process_status` test in `mcp_tools/tests/test_command_executor_async.py` is failing because it expects the process status to be `"running"` immediately after starting an async command, but the actual status returned is `"disk-sleep"`.

**Test Failure:**
```
AssertionError: assert 'disk-sleep' == 'running'
  
  - running
  + disk-sleep
```

**Root Cause:**
The `get_process_status()` method in `CommandExecutor` uses `psutil` to get detailed process information and overwrites the logical status with the OS-level process status. When a `sleep` command is executed, the process may be in a "disk-sleep" state (waiting for disk I/O) rather than the generic "running" state that the test expects.

## Proposed Solution

The `get_process_status()` method should maintain a consistent logical status model rather than exposing OS-level process states directly. We should:

1. **Preserve Logical Status**: Keep the high-level status (`"running"`, `"completed"`, `"terminated"`, etc.) separate from the detailed OS-level status
2. **Add OS Status as Separate Field**: Include the psutil status information as a separate field (e.g., `"os_status"` or `"detailed_status"`) rather than overwriting the main status
3. **Update Tests**: Ensure tests account for the various valid process states that can occur

## Acceptance Criteria

- [ ] `get_process_status()` returns consistent logical status values (`"running"`, `"completed"`, `"terminated"`, etc.)
- [ ] OS-level process status from psutil is available as a separate field (e.g., `"os_status"`)
- [ ] All existing tests pass, including `test_get_process_status`
- [ ] Tests that check for process states account for valid OS-level states like `"sleeping"`, `"disk-sleep"`, etc. where appropriate
- [ ] API backward compatibility is maintained

## Technical Details

**Affected Files:**
- `mcp_tools/command_executor/executor.py` (lines ~670-675 in `get_process_status()`)
- `mcp_tools/tests/test_command_executor_async.py` (line 119 in `test_get_process_status`)

**Current Problematic Code:**
```python
# In get_process_status() method
status_info = {
    "status": "running",  # This gets overwritten
    "pid": pid,
    "token": token,
    "command": process_data["command"],
    "runtime": time.time() - process_data["start_time"],
}

# Add psutil info if available
if process_info:
    status_info.update(process_info)  # This overwrites "status" with OS status
```

**Expected Fix:**
```python
# Preserve logical status, add OS status separately
if process_info:
    os_status = process_info.pop("status", None)  # Extract OS status
    status_info.update(process_info)
    if os_status:
        status_info["os_status"] = os_status  # Add as separate field
```

## Additional Context

This issue was discovered during CI testing on Linux where the `sleep 2` command immediately enters a "disk-sleep" state. The test assumes that any running process should report status as "running", but the current implementation exposes the granular OS-level process states.

The `query_process_no_wait` test already accounts for this by checking for multiple valid states:
```python
assert status["status"] in ["running", "sleeping", "disk-sleep"]
```

This suggests the codebase is aware of these different states, but the `get_process_status` test hasn't been updated accordingly.

**Environment:**
- OS: Linux (GitHub Actions CI)
- Python: 3.11.12
- Test Command: `sleep 2`
- Actual Status Returned: `"disk-sleep"`
- Expected Status: `"running"`

--- a/mcp_tools/command_executor/executor.py
+++ b/mcp_tools/command_executor/executor.py
@@ -672,9 +672,14 @@ class CommandExecutor(CommandExecutorInterface):
                 "runtime": time.time() - process_data["start_time"],
             }
 
-            # Add psutil info if available
+            # Add psutil info if available, but preserve logical status
             if process_info:
+                # Extract OS status before updating to avoid overwriting logical status
+                os_status = process_info.pop("status", None)
                 status_info.update(process_info)
+                # Add OS status as separate field
+                if os_status:
+                    status_info["os_status"] = os_status
 
             return status_info
 
@@ -690,9 +695,14 @@ class CommandExecutor(CommandExecutorInterface):
             "runtime": time.time() - process_data["start_time"],
         }
 
-        # Add psutil info if available
+        # Add psutil info if available, but preserve logical status
         if process_info:
+            # Extract OS status before updating to avoid overwriting logical status
+            os_status = process_info.pop("status", None)
             status_info.update(process_info)
+            # Add OS status as separate field
+            if os_status:
+                status_info["os_status"] = os_status
 
         return status_info
 

--- a/mcp_tools/command_executor/executor.py
+++ b/mcp_tools/command_executor/executor.py
@@ -605,6 +605,22 @@ class CommandExecutor(CommandExecutorInterface):
                 "error": f"Error starting command: {str(e)}",
             }
 
+    def _merge_psutil_info(self, status_info: Dict[str, Any], process_info: Optional[Dict[str, Any]]) -> None:
+        """Merge psutil process info while preserving logical status.
+
+        Args:
+            status_info: The status dictionary to update
+            process_info: Optional psutil process information
+        """
+        # Add psutil info if available, but preserve logical status
+        if process_info:
+            # Extract OS status before updating to avoid overwriting logical status
+            os_status = process_info.pop("status", None)
+            status_info.update(process_info)
+            # Add OS status as separate field
+            if os_status:
+                status_info["os_status"] = os_status
+
     async def get_process_status(self, token: str) -> Dict[str, Any]:
         """Get the status of an asynchronous process
 
@@ -672,14 +688,7 @@ class CommandExecutor(CommandExecutorInterface):
                 "runtime": time.time() - process_data["start_time"],
             }
 
-            # Add psutil info if available, but preserve logical status
-            if process_info:
-                # Extract OS status before updating to avoid overwriting logical status
-                os_status = process_info.pop("status", None)
-                status_info.update(process_info)
-                # Add OS status as separate field
-                if os_status:
-                    status_info["os_status"] = os_status
+            self._merge_psutil_info(status_info, process_info)
 
             return status_info
 
@@ -695,14 +704,7 @@ class CommandExecutor(CommandExecutorInterface):
             "runtime": time.time() - process_data["start_time"],
         }
 
-        # Add psutil info if available, but preserve logical status
-        if process_info:
-            # Extract OS status before updating to avoid overwriting logical status
-            os_status = process_info.pop("status", None)
-            status_info.update(process_info)
-            # Add OS status as separate field
-            if os_status:
-                status_info["os_status"] = os_status
+        self._merge_psutil_info(status_info, process_info)
 
         return status_info
 

